src/lib/communication/DY.Lib.Communication.RequestResponse.Lemmas.fst

```diff
@@ -234,49 +306,80 @@ val mk_comm_layer_response_nonce_proof:
   Lemma
   (requires
     trace_invariant tr /\
-    has_communication_layer_reqres_state_predicates a /\
+    has_communication_layer_reqres_state_predicate a /\
     has_communication_layer_reqres_crypto_predicates a /\
-    bytes_well_formed tr req_meta_data.key
+    bytes_well_formed tr req_meta_data.key // Can be derived from CommServerReceiveRequest event
   )
   (ensures (
     match mk_comm_layer_response_nonce req_meta_data usg tr with
     | (None, tr_out) -> trace_invariant tr_out
     | (Some nonce, tr_out) -> (
-      is_knowable_by (get_response_label tr_out req_meta_data) tr_out nonce
+      trace_invariant tr_out /\
+      is_knowable_by (get_response_label tr_out req_meta_data) tr_out nonce /\
+      get_label tr_out nonce `can_flow tr_out` get_label tr_out req_meta_data.key
```

Comment:
```suggestion
      is_knowable_by (get_label tr_out req_meta_data.key) tr_out nonce
```
(though I wonder if it's needed at all to have both of these in the postcondition, since they can be interconverted with `get_response_label_knowable` if the issue with `cinvs` can be worked out --- that would also remove the need to reveal `get_response_label` in the proof, since it could then be replaced with a lemma call).

Reply:
This edit also applies for `mk_comm_layer_response_nonce_labeled_proof`